bookkeeper-http/http-server/src/main/java/org/apache/bookkeeper/http/HttpServer.java

```diff
@@ -0,0 +1,82 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.bookkeeper.http;
+
+/**
+ * Provide the interface to start, stop bookkeeper http server.
+ * It also provide the interface to inject service provider,
+ * which is the implementation of services for http endpoints.
+ */
+public interface HttpServer {
```

Comment:
Are you suggesting adding a service to return all endpoints so that people can curl to see all possible endpoints, or just adding a function getEndpoints() in this interface? If we're just adding the function, when will we use that function?